navigator/backend/src/main/scala/com/digitalasset/navigator/json/ModelCodec.scala

```diff
@@ -99,15 +99,17 @@ object ModelCodec {
       def write(value: Contract): JsValue = JsObject(
         "id" -> value.id.toJson,
         "template" -> value.template.toJson,
-        "argument" -> value.argument.toJson
+        "argument" -> value.argument.toJson,
+        "agreementText" -> value.agreementText.toJson
       )
       def read(value: JsValue, types: DamlLfTypeLookup): Contract = {
         val id = anyField(value, "id", "Contract").convertTo[ApiTypes.ContractId]
         val template = anyField(value, "template", "Contract").convertTo[Template]
         val argument = ApiCodecCompressed
           .jsValueToApiType(anyField(value, "record", "Contract"), template.id, types)
           .asInstanceOf[ApiRecord]
-        Contract(id, template, argument)
+        val agreementText = anyField(value, "agreementText", "Contract").convertTo[Option[String]]
```

Comment:
`strField` asserts that the given field is a string field, you could use it instead of `anyField`.

Also, I'm surprised that `JsValue` (returned by `anyField`) can be magically converted to a `Option[String]`

Reply:
Fixed. I added `optionStrField` to Util.scala.